### PR TITLE
added TypeScript definition file

### PR DESCRIPTION
**Pull request by superdopey**
Added missing TypeScript definitions

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    extends React.AllHTMLAttributes<HTMLElement> {
+    errorMessages?: string[] | string
+    validators?: string[]
+    validatorListener?: () => void
+    withRequiredValidator?: boolean
+    onChange: (event: any) => void
+  }
+
+  interface ValidatorComponentState {
+    isValid
```
Comment (NewOldMax) [inline index.d.ts]: Should it be `boolean`?

Comment (superdopey) [reply index.d.ts]: What do you mean?

Comment (NewOldMax) [reply index.d.ts]: I've never used typescript, but in other places you have specified type, like
````javascript
input: any
isValid: boolean
````
should be here also
````javascript
interface ValidatorComponentState {
    isValid: boolean
}
````
?

Comment (superdopey) [reply index.d.ts]: O yeah, I missed that one. I will change it.

**Comment by NewOldMax** [comment]
Sorry, I'm not able to maintain typescript definition even if I accept this PR

**Pull request closed without merging**